### PR TITLE
Ssh

### PR DESCRIPTION
**Pull request by TravisMac**

### CODE REVIEW

File: src/ssheepdog/management/commands/update_ssh.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from django.core.management.base import BaseCommand, CommandError
+from ssheepdog.sync import test_sync
+
+class Command(BaseCommand):
```
Comment (wolfe) [inline src/ssheepdog/management/commands/update_ssh.py:4]: Remove junk

File: src/templates/base.html [hunk 2]
```diff
@@ -1,5 +1,6 @@
 <html>
 <body>
+<link rel="stylesheet" href="{{ STATIC_URL }}css/base.css"type="text/css" />
```
Comment (wolfe) [inline src/templates/base.html]: oops --- put space back in

File: src/ssheepdog/sync.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+from django.contrib.auth.models import User
+from ssheepdog.models import Login
+from fabric.api import env, sudo, run
+
+def test_sync():
+    users = User.objects.select_related('_profile_cache')
+    logins = Login.objects.all()
+    authorized_keys = []
+    for user in users:
```
Comment (wolfe) [inline src/ssheepdog/sync.py]: No need to mess with the users.  That was only nice to do for rendering a template.

File: src/ssheepdog/sync.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        user.ssh_key = user.get_profile().ssh_key
+    for login in logins:
+        for user in users:
+            if user in login.users.all():
+                m = login.machine
+                env.host_string = "%s@%s:%d" % (login.username,
+                        (m.ip or m.hostname),
+                        m.port)
+                authorized_keys.append(str(user.ssh_key))
+        print str(env.host_string)
```
Comment (wolfe) [inline src/ssheepdog/sync.py]: Left in a print

File: src/ssheepdog/sync.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+def test_sync():
+    users = User.objects.select_related('_profile_cache')
+    logins = Login.objects.all()
+    authorized_keys = []
+    for user in users:
+        user.ssh_key = user.get_profile().ssh_key
+    for login in logins:
+        for user in users:
+            if user in login.users.all():
```
Comment (wolfe) [inline src/ssheepdog/sync.py]: Need is_active checks

File: src/ssheepdog/sync.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    logins = Login.objects.all()
+    authorized_keys = []
+    for login in logins:
+        for user in users:
+            if user in login.users.all():
+                m = login.machine
+                env.host_string = "%s@%s:%d" % (login.username,
+                        (m.ip or m.hostname),
+                        m.port)
+                authorized_keys.append(str(user.get_profile().ssh_key))
```
Comment (wolfe) [inline src/ssheepdog/sync.py]: str is redundant

File: src/ssheepdog/sync.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
+from django.contrib.auth.models import User
+from ssheepdog.models import Login
+from fabric.api import env, sudo, run
+
+def test_sync():
+    users = User.objects.select_related('_profile_cache')
+    logins = Login.objects.all()
+    authorized_keys = []
```
Comment (wolfe) [inline src/ssheepdog/sync.py]: You  need a different authorized_keys list for each login

File: src/ssheepdog/sync.py [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+    for login in logins:
+        for user in users:
+            if user in login.users.all():
```
Comment (wolfe) [inline src/ssheepdog/sync.py]: Your list should only include those users who are active

File: src/ssheepdog/sync.py [hunk 9]
```diff
@@ -0,0 +1,17 @@
+from django.contrib.auth.models import User
+from ssheepdog.models import Login
+from fabric.api import env, sudo, run
+
+def test_sync():
+    users = User.objects.select_related('_profile_cache')
+    logins = Login.objects.all()
+    authorized_keys = []
+    for login in logins:
```
Comment (wolfe) [inline src/ssheepdog/sync.py:8]: If a login is not active, you need to push out no users

File: src/ssheepdog/sync.py (hunk 6 above)
Comment (wolfe) [inline src/ssheepdog/sync.py]: There should be a member function of a login get_keys() which does this the work of going through all the active users and ...

File: src/templates/base.html [hunk 10]
```diff
@@ -1,5 +1,6 @@
 <html>
 <body>
+<link rel="stylesheet" href="{{ STATIC_URL }}css/base.css" type="text/css" />
```
Comment (thurloat) [inline src/templates/base.html]: better that it's in the base template, but css is supposed to be in the head tag.

```
<html>
    <head>
        <link ... />        <-- css declaration goes in head tag.
    </head>
    <body>
    </body>
</head>
```

**Pull request merged**